pypiserver/backend.py

```diff
@@ -167,6 +167,15 @@ def __init__(
 
         self.cache_manager = cache_manager or CacheManager()  # type: ignore
 
+    def add_package(self, filename: str, stream: t.BinaryIO) -> None:
+        write_file(stream, self.roots[0].joinpath(filename))
+        self.cache_manager.invalidate_root_cache(self.roots[0])
+
+    def remove_package(self, pkg: PkgFile) -> None:
+        if pkg.fn is not None:
+            os.remove(pkg.fn)
```

Comment:
I think we need to catch `FileNotFoundError` here to be safe. (Or maybe even `OSError`, although that might be too broad a catch)

Reply:
So frustrating that python doesn't include which exceptions can be raised from a function in the stdlib documentation honestly

Reply:
Yeah I know, but I guess that'd be documentation that's very hard to maintain if it needs to list all possible exceptions.

also: I meant `IOError` earlier, not `OSError`. It's still probably to broad an exception group to (silently) handle

Reply:
I added in a catch of `FileNotFoundError` with a log message and a pass, and a catch of `OSError`, which logs an exception and then re-raises :shrug: